src/meshFeature.js

```diff
@@ -0,0 +1,444 @@
+    }
+    if (value === undefined) {
+      var mesh = $.extend(
+        {},
+        {
+          gridWidth: function () {
+            if (specOrProperty.gridHeight) {
+              return Math.floor(m_this.data().length / specOrProperty.gridHeight);
+            }
+            return Math.floor(Math.sqrt(m_this.data().length));
+          },
+          gridHeight: function () {
+            if (specOrProperty.gridWidth) {
+              return Math.floor(m_this.data().length / specOrProperty.gridWidth);
+            }
+            return Math.floor(Math.sqrt(m_this.data().length));
+          }
+        },
+        m_mesh,
+        specOrProperty
+      );
+      m_mesh = mesh;
+    } else {
+      m_mesh[specOrProperty] = value;
+    }
+    m_this.modified();
+    return m_this;
+  };
+
+  /**
+   * A uniform getter that always returns a function even for constant values.
+   * If undefined input, return all the mesh values as an object.
+   *
+   * @param {string|undefined} key The name of the mesh key or `undefined` to
+   *    return an object with all keys as functions.
+   * @returns {object|function} A function related to the key, or an object
+   *    with all mesh keys, each of which is a function.
+   */
+  this.mesh.get = function (key) {
+    if (key === undefined) {
+      var all = {}, k;
+      for (k in m_mesh) {
+        if (m_mesh.hasOwnProperty(k)) {
+          all[k] = m_this.mesh.get(k);
+        }
+      }
+      return all;
+    }
+    return util.ensureFunction(m_mesh[key]);
+  };
+
+  /**
+   * Get/Set position accessor.  This is identical to getting or setting the
+   * `position` style.
+   *
+   * @param {array|function} [val] If specified, set the position style.  If
+   *    `undefined`, return the current value.
+   * @returns {array|function|this} Either the position style or this.
+   */
+  this.position = function (val) {
+    if (val === undefined) {
+      return m_this.style('position');
+    } else {
+      m_this.style('position', val);
+      m_this.dataTime().modified();
+      m_this.modified();
+    }
+    return m_this;
+  };
+
+  /**
+   * Create a set of vertices and elements from a mesh specification.
+   *
+   * This currently takes a set of `vertexValueFuncs` to detemrine which
+   * vertices are used (shown) and values at those vertices.  It could be
+   * extended with `elementValueFuncs`.
+   *
+   * @param {object} [vertexValueFuncs] A dictionary where the keys are the
+   *    names of properties to include in the results and the values are
+   *    functions that are evaluated at each vertex with the arguments
+   *    (data[idx], idx).  If a key is named `used`, then if its function
+   *    returns a falsy value for a data point, the vertex associated with that
+   *    data point is removed from the resultant mesh.
+   * @returns {geo.meshFeature.meshInfo} An object with the mesh information.
+   */
+  this.createMesh = function (vertexValueFuncs) {
```

Comment:
@manthey does this needs to be in public API?

Reply:
It doesn't need to be.  My reasoning for it to be a public API, is that making so encourages its reuse in other components.

Reply:
I see but from the end-user perspective, I do see rare use specially since this does not seem like a self contained method. I would make it _createMesh but if you have strong feelings about keeping it public, I am fine with that.

Reply:
I'll change it.